src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -438,12 +440,32 @@ public static TestPackage MakeTestPackage(ConsoleOptions options)
             if (options.DefaultTestNamePattern != null)
                 package.AddSetting(FrameworkPackageSettings.DefaultTestNamePattern, options.DefaultTestNamePattern);
 
-            if (options.TestParameters != null)
-                package.AddSetting(FrameworkPackageSettings.TestParameters, options.TestParameters);
+            if (options.TestParameters.Count != 0)
+                AddTestParametersSetting(package, options.TestParameters);
 
             return package;
         }
 
+        /// <summary>
+        /// Sets test parameters, handling backwards compatibility.
+        /// </summary>
+        private static void AddTestParametersSetting(TestPackage testPackage, IDictionary<string, string> testParameters)
+        {
+            testPackage.AddSetting(FrameworkPackageSettings.TestParametersDictionary, testParameters);
+
+            if (testParameters.Count != 0)
+            {
+                // This cannot be changed without breaking backwards compatibility with old frameworks.
+                // Reserializes the way old frameworks understand, even if this runner's parsing is changed.
+
+                var oldFrameworkSerializedParameters = new StringBuilder();
+                foreach (var parameter in testParameters)
+                    oldFrameworkSerializedParameters.Append(parameter.Key).Append('=').Append(parameter.Value).Append(';');
+
+                testPackage.AddSetting(FrameworkPackageSettings.TestParameters, oldFrameworkSerializedParameters.ToString(0, oldFrameworkSerializedParameters.Length - 1));
```

Comment:
This potentially changes the order in which the parameters appear in the string, since the dictionary is free to return the pairs in any order. The original implementation created the string in the same order as they appeared on the command-line. This may or may not be an issue.